float64[] params_s
float64[] params_e
